ros2_ws/Dockerfile

```diff
@@ -1,152 +1,10 @@
-RUN mkdir -p ${COLCON_HOME}
-COPY --chown=${USER}:${USER} ./config/colcon ${COLCON_HOME}
-RUN /bin/bash ${COLCON_HOME}/setup.sh
-
-# build ROS workspace
-RUN mkdir -p ${ROS2_WORKSPACE}/src
-WORKDIR ${ROS2_WORKSPACE}
-RUN rosdep update
-RUN /bin/bash -c "source /opt/ros/${ROS_DISTRO}/setup.bash; colcon build --symlink-install"
-
-# prepend the environment sourcing to bashrc (appending will fail for non-interactive sessions)
-RUN echo "source /opt/ros/${ROS_DISTRO}/setup.bash; \
-source ${ROS2_WORKSPACE}/install/setup.bash" | cat - ${HOME}/.bashrc > tmp && mv tmp ${HOME}/.bashrc
-# enable colorized output from ros logging
-RUN echo "export RCUTILS_COLORIZED_OUTPUT=1" >> ${HOME}/.bashrc
-
-# create the credentials to be able to pull private repos using ssh
-USER root
-RUN mkdir /root/.ssh/ && ssh-keyscan github.com | tee -a /root/.ssh/known_hosts
-RUN echo "session required pam_limits.so" | tee --append /etc/pam.d/common-session > /dev/null
-
-FROM base-workspace as ros2-control-iron
-
-RUN apt-get update && apt-get install --no-install-recommends -y \
-  doxygen \
-  librange-v3-dev \
-  python3-jinja2 \
-  python3-typeguard \
-  && rm -rf /var/lib/apt/lists/*
-
-# build as ROS user
-USER ${USER}
-WORKDIR ${ROS2_WORKSPACE}/src
-
-RUN git clone -b 3.18.0 --depth 1 https://github.com/ros-controls/ros2_control.git
-RUN git clone -b 3.14.0 --depth 1 https://github.com/ros-controls/ros2_controllers.git
-# get additional interface dependencies manually
-RUN git clone -b 5.0.0 --depth 1 https://github.com/ros-controls/control_msgs.git
-RUN git clone -b 1.0.2 --depth 1 https://github.com/pal-robotics/backward_ros.git
-RUN git clone -b 3.1.2 --depth 1 https://github.com/ros/diagnostics.git
-RUN git clone -b 2.5.0 --depth 1 https://github.com/ros-controls/realtime_tools.git
-RUN git clone -b 1.7.0 --depth 1 https://github.com/ros2/rcl_interfaces.git
-RUN git clone -b 0.11.0 --depth 1 https://github.com/ros2/test_interface_files.git
-RUN git clone -b 1.16.0 --depth 1 https://github.com/ros/angles.git
-RUN git clone -b ros2 --depth 1 https://github.com/ros-drivers/ackermann_msgs.git
-RUN git clone -b 0.3.3 --depth 1 https://github.com/PickNikRobotics/generate_parameter_library.git
-RUN git clone -b 0.2.2 --depth 1 https://github.com/PickNikRobotics/RSL.git
-RUN git clone -b 1.0.2 --depth 1 https://github.com/PickNikRobotics/cpp_polyfills.git
-RUN git clone -b 3.1.0 --depth 1 https://github.com/ros-controls/control_toolbox.git
-RUN git clone -b 0.1.0 --depth 1 https://github.com/ros-controls/kinematics_interface.git
-
-WORKDIR ${ROS2_WORKSPACE}
-RUN /bin/bash -c "source /opt/ros/${ROS_DISTRO}/setup.bash; colcon build"
-RUN rm -rf ./src
-
-FROM ros2-control-${BASE_TAG} as ros2-control
-
-FROM environment-variables as final
-COPY --from=ros2-control / /
-
-# start as ROS user
-USER ${USER}
-WORKDIR ${ROS2_WORKSPACE}
-
-# Metadata
-ARG VERSION
-ARG BASE_TAG
+ARG VERSION=v0.0.0
```

Comment:
Just a question without going into the CI workflow to check myself: the VERSION.iron is written as `1.0.0` (not `v1.0.0`) but here the default ARG includes the v prefix `v0.0.0` (which is something I just wrote for testing). Should the `org.opencontainers.image.version="${VERSION}"` include the v or not? I would just prefer to double check that the default format of the ARG is representative of the intended metadata format, since that helps readability and avoids confusion.

Reply:
The version that the CI action from our `.github` repo gives is in the format `vX.Y.Z`. We never put the `v` in our `VERSION` files, since the package.xml also doesn't contain it. However, if Louis wants the `v` in the metadata, I can't be 100% sure but I suppose so since he made those changes himself.

Reply:
For future reference:

* For git labels: I find it convenient to have them as it does better grouping of the tags
* For anything we parse (aica-pacakage.toml, metadata): with or without doesn't matter as our parser handle either (https://github.com/hashicorp/go-version/blob/v1.6.0/version.go#L21)